modules/ROOT/pages/salesforce/salesforce-connector-96.adoc

```diff
@@ -0,0 +1,1211 @@
+		</ee:transform>
+	</flow>
+	<flow name="update_accountFlow" >
+		<http:listener config-ref="HTTP_Listener_config" path="/update" doc:name="Listener"  />
+		<ee:transform doc:name="Transform Message"  >
+			<ee:message >
+				<ee:set-payload ><![CDATA[%dw 2.0
+output application/java
+---
+[{
+
+	Name: payload.Name,
+	AccountNumber: payload.AccountNumber,
+	Id:payload.Id
+}]]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+		<salesforce:update config-ref="Salesforce_Sfdc_config" type="Account" doc:name="Update"  />
+		<ee:transform doc:name="Transform Message" >
+			<ee:message >
+				<ee:set-payload ><![CDATA[%dw 2.0
+output application/json
+---
+payload map {
+	id:$.id,
+	errors:$.errors,
+	success:$.success
+}]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+	</flow>
+	<flow name="upsert_accountFlow" >
+		<http:listener config-ref="HTTP_Listener_config" path="/upsert" doc:name="Listener" />
+		<ee:transform doc:name="Transform Message" >
+			<ee:message >
+				<ee:set-payload ><![CDATA[%dw 2.0
+output application/java
+---
+[{
+
+	Name: payload.Name,
+	AccountNumber: payload.AccountNumber,
+	Id:payload.Id
+}]]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+		<salesforce:upsert config-ref="Salesforce_Sfdc_config"
+		externalIdFieldName="Id" type="Account" doc:name="Upsert" />
+		<ee:transform doc:name="Transform Message" >
+			<ee:message >
+				<ee:set-payload ><![CDATA[%dw 2.0
+output application/json
+---
+payload map {
+	id:$.id,
+	errors:$.errors,
+	success:$.success,
+	created:$.created
+
+	}]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+	</flow>
+	<flow name="find_duplicates_for_account_flow" >
+		<http:listener config-ref="HTTP_Listener_config" path="/findDuplicates" doc:name="Listener" />
+		<ee:transform doc:name="Transform Message" >
+			<ee:message >
+				<ee:set-payload ><![CDATA[%dw 2.0
+output application/java
+---
+[
+	payload
+]]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+		<salesforce:find-duplicates config-ref="Salesforce_Sfdc_config" type="Account"
+		doc:name="Find duplicates" />
+		<ee:transform doc:name="Transform Message" >
+			<ee:message >
+				<ee:set-payload ><![CDATA[%dw 2.0
+output application/json
+---
+{
+	success: payload.success,
+	duplicateResults: {
+		(payload.duplicateResults map {
+			matchRecords: $.matchResults
+		}
+		)
+	},
+	duplicateRuleEntityType: payload.duplicateRuleEntityType,
+	duplicateRule: payload.duplicateRule,
+	allowSave: payload.allowSave,
+	errorMessage: payload.errorMessage
+}]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+	</flow>
+	<flow name="crud-appFlow" >
+		<http:listener doc:name="Listener" config-ref="HTTP_Listener_config" path="/"/>
+		<salesforce:convert-lead doc:name="Convert lead" config-ref="Salesforce_Sfdc_config"/>
+	</flow>
+</mule>
+----
+
+== Example: Create or Update an sObject With Parent Child Relationships
+
+A Salesforce object (sObject) can have standard or custom relationships between objects.
```

Comment:
only sObject is an sObject--most of the objects are instances of it, and just called Account object or Contact object etc...